android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/CloudDataJsonUtil.java

```diff
@@ -75,13 +76,14 @@ private CloudDataJsonUtil()
         int kbLength = langKeyboards.length();
         for (int j = 0; j < kbLength; j++) {
           JSONObject keyboardJSON = langKeyboards.getJSONObject(j);
+          String pkgID = keyboardJSON.optString(KMManager.KMKey_PackageID, KMManager.KMDefault_UndefinedPackageID);
           String kbID = keyboardJSON.getString(KMManager.KMKey_ID);
           String kbName = keyboardJSON.getString(KMManager.KMKey_Name);
           String kbVersion = keyboardJSON.optString(KMManager.KMKey_KeyboardVersion, "1.0");
           String kbFont = keyboardJSON.optString(KMManager.KMKey_Font, "");
 
           //String kbKey = String.format("%s_%s", langID, kbID);
-          HashMap<String, String> hashMap = createKeyboardInfoMap(null,langID,langName,kbID,kbName,kbVersion,isCustom,kbFont,null);
+          HashMap<String, String> hashMap = createKeyboardInfoMap(pkgID,langID,langName,kbID,kbName,kbVersion,isCustom,kbFont,null);
```

Comment:
`version` and `modelVersion` in lines 140-141 were already handled by previous 13.0 development.